model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/ContentExplorer.kt

```diff
@@ -112,14 +119,27 @@ class ContentExplorer(private val client: IModelClient, private val repoManager:
                 repoManager.runWithRepository(repositoryId) {
                     val tree = CLVersion.loadFromHash(versionHash, client.storeCache).getTree()
                     val rootNode = PNodeAdapter(ITree.ROOT_ID, TreePointer(tree))
+                    val expandTo = call.request.queryParameters["expandTo"]?.toLongOrNull()
```

Comment:
```suggestion
                    // IMPORTANT Do not let `expandTo` be an arbitrary string to avoid code injection.
                    // The value `expandTo` is expanded into JavaScript.
                    val expandTo = call.request.queryParameters["expandTo"]?.toLongOrNull()
```

Good, that you called `toLongOrNull` on `call.request.queryParameters["expandTo"]`.
Else `expandTo` could have been used to inject arbitrary JS code in `scrollToElement('$expandTo');`.

Reply:
A different option would be, to set the [URI fragment](https://developer.mozilla.org/en-US/docs/Web/URI/Fragment) to let the browser make the scrolling automatic.

See https://stackoverflow.com/a/13542294

Reply:
I didn't like having the nodeId visible to the user both as a GET parameter and a fragment. That's why I opted for the JS solution.

Reply:
done